packages/source-commonmark/src/index.ts

```diff
@@ -191,14 +191,26 @@ export class Parser {
 
 export default class extends Document {
   constructor(markdown: string) {
-    let md = MarkdownIt('commonmark');
-    let parser = new Parser(md.parse(markdown, { linkify: false }), {});
     super({
-      content: parser.content,
+      content: '',
       contentType: 'text/commonmark',
-      annotations: parser.annotations,
+      annotations: [],
       schema: markdownSchema
     });
+
+    let md = this.constructor.markdownParser();
+    let parser = new Parser(md.parse(markdown, { linkify: false }), this.constructor.contentHandlers());
+
+    this.content = parser.content;
+    this.annotations = parser.annotations;
+  }
+
+  static markdownParser() {
+    return MarkdownIt('commonmark');
+  }
+
+  static contentHandlers() {
```

Comment:
Would it make sense to explode this out so each type of content is handled in a different function?

cf.

```js
static underline(token: MarkdownIt.Token) {}
```

Or would it make more sense for this to be on the prototype?

```js
underline(token: MarkdownIt.Token) {}
```

Reply:
trival - worth adding instance getters as well?
```js
get parser() {
  return this.constructor.markdownParser();
}
```
i dunno

Reply:
@tim-evans per conversation in slack, I think having this be defined as an object that's returned from the subclass is fine because it gives a little more flexibility to share methods piecemeal, which will probably come in handy (i.e., not all brands will support all features, so compartmentalising them and linking via object properties probably makes the most sense?)

@pgoldrbx I can't think of a case where we'd want those to be used - is there a specific use-case you have in mind? I'm not opposed, just erring on keeping the interface surface small.

Reply:
no not at all - it's probably fine - like i said, i don't even know the intended use cases!